common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java

```diff
@@ -230,6 +245,101 @@ protected final void setAuthorizationEndpoint(final String authorizationEndpoint
+        final String methodName = ":getDeviceCode";
+
+        // Set up connection
+        String urlBody;
+
+        if (authorityUrl == null) {
+            // Fetch the tenant from the authority
+            String tenant = authorizationRequest.getAuthority().getPath().split("/")[1];
+
+            // Set up Device Code Flow authorization url
+            urlBody = "https://login.microsoftonline.com/" + tenant + "/oauth2/v2.0/devicecode";
+        }
+        else {
+            urlBody = authorityUrl;
+        }
+
+        URL url = new URL(urlBody);
+
+        final HttpURLConnection con = (HttpsURLConnection) url.openConnection();
+        con.setRequestMethod("POST");
+        con.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+        con.setConnectTimeout(5000);
+        con.setReadTimeout(5000);
+
+        // Set up request body
+        con.setDoOutput(true);
+        final DataOutputStream out = new DataOutputStream(con.getOutputStream());
+        StringBuilder encodedResult = new StringBuilder();
+
+        // Encode client_id
+        encodedResult.append(URLEncoder.encode("client_id", "UTF-8"));
+        encodedResult.append("=");
+        encodedResult.append(URLEncoder.encode(authorizationRequest.getClientId(), "UTF-8"));
+        encodedResult.append("&");
+
+        // Encode scope
+        encodedResult.append(URLEncoder.encode("scope", "UTF-8"));
+        encodedResult.append("=");
+        encodedResult.append(URLEncoder.encode(authorizationRequest.getScope(), "UTF-8"));
+
+        // Attach request body
+        final String convertedParams = encodedResult.toString();
+        out.writeBytes(convertedParams);
+        out.flush();
+        out.close();
+
+        // Send request and create the authorization result
+        // MicrosoftSTAuthorizationResultFactory not used since no Intent is being created
+        final int authorizationCode = con.getResponseCode();
+        final BufferedReader streamReader;
+        final AuthorizationResult authorizationResult;
+
+        // Check if the request was successful
+        // Any code below 300 (HTTP_MULT_CHOICE) is considered a success
+        if (authorizationCode < HttpsURLConnection.HTTP_MULT_CHOICE){
+            // Get and parse response body
+            streamReader = new BufferedReader(new InputStreamReader(con.getInputStream()));
+            final String responseBody = streamReader.readLine();
+            final HashMap<String, String> parsedResponseBody = new Gson().fromJson(responseBody, new TypeToken<HashMap<String, String>>() {}.getType());
+
+            // Create response and result objects
+            // Not sure where "code" is stored. Not found in the input stream...
+            final MicrosoftStsAuthorizationResponse authorizationResponse =
+                    new MicrosoftStsAuthorizationResponse(null, authorizationRequest.getState(), parsedResponseBody);
```

Comment:
Any ideas where this would be found?

Reply:
This looks to be referring to the `auth_code` normally returned by the Authorization Code Flow. I think you can leave null if this is unused in your scenario and if the field is allowed to be nullable

Reply:
It's nullable, but the JavaDoc for the field states that it's required. This may just be a remnant of the old code back when only auth_code was supported.

Reply:
Since you won't have this value, I'll suggest we add a new constructor to use for DCF flows. If that ends up messy (or complicates the code), feel free to update the docs and pass `null`